metadata_service/api/tag.py

```diff
@@ -58,16 +58,12 @@ def put(self, id: str, resource_type: ResourceType,
 
         whitelist_badges = app.config.get('WHITELIST_BADGES', [])
         if tag_type == BADGE_TYPE:
-            # need to check whether the badge is part of the whitelist:
-            if tag not in whitelist_badges:
-                return \
-                    {'message': 'The tag {} for id {} with type {} and resource_type {} '
-                                'is not added successfully as badge '
-                                'is not part of the whitelist'.format(tag,
-                                                                      id,
-                                                                      tag_type,
-                                                                      resource_type.name)}, \
-                    HTTPStatus.NOT_FOUND
+            # TODO remove adding badge functionality, still don't allow tags to have badge names
+            # TODO update unit tests
+            return \
+                {'message': 'Badges should be added using /badges/, tag_type=badge no longer valid'}, \
+                HTTPStatus.NOT_ACCEPTABLE
+
         else:
             if tag in whitelist_badges:
```

Comment:
if we use dict type for badges, I don't think this line will work. So if an user create a tag with badge name, they could do so now...

Reply:
oh that's right I made the change before I changed to using dicts, I can change this to be Badge and find by badge.badge_name